modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/swap/DiscountingNotionalExchangePricer.java

```diff
@@ -83,4 +84,18 @@ public void explainPresentValue(NotionalExchange event, RatesProvider provider,
     }
   }
 
+  //-------------------------------------------------------------------------
+  @Override
+  public MultiCurrencyAmount currencyExposure(NotionalExchange event, RatesProvider provider) {
+    return MultiCurrencyAmount.of(CurrencyAmount.of(event.getCurrency(), presentValue(event, provider)));
+  }
+
+  @Override
+  public double currentCash(NotionalExchange event, RatesProvider provider) {
```

Comment:
Same remark as above: As the amount is known, we don't need the full provider, the valuation date is enough.